java/src/main/native/src/ColumnViewJni.cpp

```diff
@@ -1760,4 +1763,50 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnView_copyColumnViewToCV(JNIEnv
   }
   CATCH_STD(env, 0)
 }
+
+JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnView_replaceChildrenWithViews(
+    JNIEnv *env, jobject j_object, jlong j_handle, jintArray j_indices, jlongArray j_children) {
+
+  JNI_NULL_CHECK(env, j_handle, "native handle is null", 0);
+  JNI_NULL_CHECK(env, j_indices, "child indices to replace can't be null", 0);
+  JNI_NULL_CHECK(env, j_children, "children to replace can't be null", 0);
+
+  try {
+    cudf::jni::native_jpointerArray<cudf::column_view> children_to_replace(env, j_children);
+    cudf::jni::native_jintArray indices(env, j_indices);
+    JNI_ARG_CHECK(env, indices.size() == children_to_replace.size(), "The indices size and children size should match", 0);
+
+    cudf::column_view *n_col_view = reinterpret_cast<cudf::column_view *>(j_handle);
+    cudf::type_id id = n_col_view->type().id();
+
+    std::map<int32_t, cudf::column_view*> m;
+    for (int i = 0 ; i < indices.size() ; i++) {
+      auto it = m.find(indices[i]);
+      JNI_ARG_CHECK(env, it == m.end(), "Duplicate mapping found for replacing child index", 0);
+      m[indices[i]] = children_to_replace[i];
+    }
+
+    std::vector<cudf::column_view> children;
+    children.reserve(n_col_view->num_children());
+    int j = 0;
+    for (int i = 0 ; i < n_col_view->num_children() ; i++) {
+      auto it = m.find(i);
+      if (it != m.end()) {
+        JNI_ARG_CHECK(env, (*it->second).size() == n_col_view->child(i).size(), "Child size don't match", 0);
+        m.erase(it);
+        children.emplace_back(*it->second);
```

Comment:
I'm not convinced that this is safe, depending on implementation. This dereferences the iterator `it` after the element it points to has been erased from the map.
Would you consider swapping the order?
```c++
        children.emplace_back(*it->second);
        m.erase(it);
```

Reply:
This is no longer a problem: `Java_ai_rapids_cudf_ColumnView_replaceChildrenWithViews` has been obviated.